src/mscorlib/corefx/Interop/Unix/System.Globalization.Native/Interop.Collation.cs

```diff
@@ -25,7 +25,7 @@ internal static partial class GlobalizationInterop
 
         [SecurityCritical]
         [DllImport(Libraries.GlobalizationInterop, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IndexOf")]
-        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options);
+        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options, ref int matchLength);
```

Comment:
I went with int* suggestion, int* reads better as it describes the actual type of the pointer, and I need to manipulate the pointer on managed side as well some times (for OrdinalCompare cases).